Source/controls/touch/event_handlers.cpp

```diff
@@ -0,0 +1,259 @@
+#ifndef USE_SDL1
+
+#include "controls/touch/event_handlers.h"
+
+#include "control.h"
+#include "diablo.h"
+#include "engine.h"
+#include "gmenu.h"
+#include "scrollrt.h"
+#include "stores.h"
+#include "utils/ui_fwd.h"
+
+namespace devilution {
+
+namespace {
+
+VirtualGamepadEventHandler Handler(&VirtualGamepadState);
+
+Point ScaleToScreenCoordinates(float x, float y)
+{
+	return Point {
+		(int)round(x * gnScreenWidth),
+		(int)round(y * gnScreenHeight)

```

Comment:
Do we know if boarders are subtracted from this by SDL, or will we need to take that in to account?

Reply:
SDL seems to take care of it. Tested on both Pixel 4 and a touchscreen Windows laptop.